Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionMatchPatternCocoa.mm

```diff
@@ -109,7 +109,7 @@
 RefPtr<WebExtensionMatchPattern> WebExtensionMatchPattern::getOrCreate(const String& scheme, const String& host, const String& path)
 {
     String resolvedScheme = !scheme.isEmpty() ? scheme : "*"_s;
-    String resolvedHost = !host.isEmpty() ? host : "*"_s;
+    String resolvedHost = !host.isEmpty() || equalLettersIgnoringASCIICase(scheme, "file"_s) ? host : "*"_s;
```

Comment:
Nit - As this logic is now slightly more complex, I wonder if this should be refactored into a separate function since we do this in more than one place?

Reply:
It is subtly different in both places, unfortunately.